version: '3.8'

services:
  db_test:
    container_name: sm_db_test
    image: postgres:16.2
    env_file:
      - .env.test
    environment:
      - POSTGRES_DB=${DB_NAME}
      - POSTGRES_USER=${DB_USER}
      - POSTGRES_PASSWORD=${DB_PASS}
    ports:
      - ${DB_PORT}:5432
    restart: on-failure
    healthcheck:
      test: ['CMD-SHELL', 'pg_isready -U ${DB_USER}']
      interval: 10s
      timeout: 5s
      retries: 5
    networks:
      - sm_backend
  tests:
    build:
      context: .
      dockerfile: Dockerfile.test
    container_name: sm_tests
    depends_on:
      db_test:
        condition: service_healthy
    ports:
      - '8000:8000'
    env_file:
      - .env.test
    volumes:
      - ./:/sm_backend/
    networks:
      - sm_backend

networks:
  sm_backend:
    driver: "bridge"
